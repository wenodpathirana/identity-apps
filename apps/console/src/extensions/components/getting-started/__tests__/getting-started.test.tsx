/**
 * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.com). All Rights Reserved.
 *
 * This software is the property of WSO2 Inc. and its suppliers, if any.
 * Dissemination of any information or reproduction of any material contained
 * herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
 * You may not alter or remove any copyright or other notice from copies of this content."
 */

import { render, screen, waitFor } from "@unit-testing";
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { fullPermissions } from "./__mocks__/getting-started-permissions";
import GettingStartedPage from "../getting-started";

describe("Test if the Getting Started page is working as expected", () => {

    it("<GettingStartedPage /> matches snapshot", () => {
        const { container } = render(<GettingStartedPage />, {
            allowedScopes: fullPermissions
        });

        expect(container).toMatchSnapshot();
    });

    it("<GettingStartedPage /> renders without exploding", () => {
        render(<GettingStartedPage />, {
            allowedScopes: fullPermissions
        });

        expect(screen.getByTestId("getting-started-page-layout")).toBeInTheDocument();
    });

    it.skip("<GettingStartedPage /> renders loaders properly", async () => {
        render(<GettingStartedPage />, {
            allowedScopes: fullPermissions
        });

        expect(screen.getByTestId(
            "dynamic-application-context-card-navigate-to-application-list-button"
        )).toBeInTheDocument();

        expect(screen.getByTestId("getting-started-page-layout")).toBeInTheDocument();
        expect(screen.getByTestId("getting-started-page-add-user-button")).toHaveClass("ui loading");
        expect(screen.getByTestId("develop-getting-started-page-add-social-login")).toBeInTheDocument();
        expect(screen.getByTestId("develop-getting-started-page-view-docs")).toBeInTheDocument();
    });
});
