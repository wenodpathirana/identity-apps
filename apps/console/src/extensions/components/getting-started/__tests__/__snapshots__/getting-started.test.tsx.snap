// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the Getting Started page is working as expected <GettingStartedPage /> matches snapshot 1`] = `
<div>
   
  <div
    class="layout page-layout getting-started-page"
    data-componentid="getting-started-page-layout"
    data-testid="page-layout"
  >
    <div
      class="layout-content"
    >
      <div
        class="loaders content-loader"
        data-componentid="content-loader-wrapper"
        data-testid="content-loader-wrapper"
      >
        <div
          class="ui active transition visible inverted dimmer"
          data-componentid="content-loader-dimmer"
          data-testid="content-loader-dimmer"
          style="display: flex;"
        >
          <div
            class="content"
          >
            <div
              class="ui inverted loader"
              data-componentid="content-loader"
              data-testid="content-loader"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
   
</div>
`;
